clean up references of cluster auto pause (#4331)
camunda/camunda-docs: docs/components/concepts/clusters.md

### Auto-pause

:::caution
**Cluster auto-pause** is not yet available and only applies to non-Enterprise clusters.
:::

> 

Reviewer: Shouldn't this caution be deleted as auto-pause is out now?

Free Trial `dev` (or untagged) clusters are automatically paused eight hours after a cluster is created or resumed from a paused state. Auto-pause occurs regardless of cluster usage.

You can resume a paused cluster at any time, which typically takes five to ten minutes to complete. See [resume your cluster](/components/console/manage-clusters/manage-cluster.md#resume-a-cluster).

- Clusters tagged as `test`, `stage`, or `prod` do not auto-pause.
- Paused clusters are automatically deleted after 30 consecutive paused days. You can change the tag to avoid cluster deletion.
- No data is lost while a cluster is paused. All execution and configuration is saved, but cluster components such as Zeebe and Operate are temporarily disabled until you resume the cluster.

> - When a cluster is paused, not all functionality is limited. For example, you may still execute BPMN timers and BPMN message catch events.

Reviewer: ```suggestion
- Paused clusters still retain some functionality. For example, you can still execute BPMN timers and BPMN message catch events.
```
Rewording suggestion
